staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -172,15 +173,15 @@ func (cnc *CloudNodeController) Run(stopCh <-chan struct{}, controllerManagerMet
 		return
 	}
 
-	// The periodic loop for updateNodeStatus communicates with the APIServer with a worst case complexity
-	// of O(num_nodes) per cycle. These functions are justified here because these events fire
-	// very infrequently. DO NOT MODIFY this to perform frequent operations.
+	// The periodic loop for updateNodeStatus polls the Cloud Provider periodically
+	// to reconcile the nodes addresses and labels.
 	go wait.Until(func() {
 		if err := cnc.UpdateNodeStatus(context.TODO()); err != nil {
 			klog.Errorf("failed to update node status: %v", err)
 		}
 	}, cnc.nodeStatusUpdateFrequency, stopCh)
 
+	// This is only used during bootstrapping
```

Comment:
Please correct me if I'm wrong, but I don't think it's really true.
This is invoke on any node addition to the cluster, which may happen outside of bootstrapping.

Reply:
yes, let me add a better comment, this is for new added nodes that are tainined